File: app/producer.js
const kafka = require('kafka-node');

const client = new kafka.KafkaClient({ kafkaHost: 'kafka:9092' });
const producer = new kafka.Producer(client);

producer.on('ready', () => {
  console.log('Productor listo para enviar mensajes');
});

producer.on('error', (err) => {
  console.error('Error en el productor', err);
});

const sendMessage = (message) => {
  const payloads = [
    { topic: 'test', messages: message }
  ];
  producer.send(payloads, (err, data) => {
    if (err) {
      console.error('Error enviando mensaje:', err);
    } else {
      console.log('Mensaje enviado:', data);
    }
  });
};

module.exports = { sendMessage };

File: app/producer.py
from kafka import KafkaProducer
import time

producer = KafkaProducer(bootstrap_servers='kafka:9092')

while True:
    producer.send('my_topic', b'Hola mundo de, Kafka!')
    print("Mensaje enviado")
    time.sleep(5)  # Espera 5 segundos entre mensajes
